sdk/core/core/inc/az_span.h

```diff
@@ -226,213 +185,186 @@ AZ_NODISCARD bool az_span_is_content_equal_ignoring_case(az_span span1, az_span
  * @return The position of `target` in `source` if `source` contains `target`,
- *         0 if `target` is empty (if its length is equal zero),
- *         -1 if `source` is empty (if its length is equal zero) and `target` is non-empty,
+ *         0 if `target` is empty (if its size is equal zero),
+ *         -1 if `source` is empty (if its size is equal zero) and `target` is non-empty,
  *         -1 if `target` is not found in `source`.
  */
 AZ_NODISCARD int32_t az_span_find(az_span source, az_span target);
 
-
-/******************************  SPAN APPENDING */
+/******************************  SPAN COPYING */
 
 /**
- * @brief az_span_append_uint8 appends the uint8 \p byte to the \p destination starting at the
- * destination span length.
+ * @brief az_span_copy copies the content of the source span to the destination span.
  *
- * @param[in] destination The az_span where the byte should be appended to.
- * @param[in] byte The uint8 to append to the destination span
- * @return An #az_span that is a clone of the \p destination span with its length increased by 1.
- *         The method assumes that the \p destination has a large enough capacity to hold one more
- * byte.
+ * @param[in] destination The span whose bytes will be replaced by the source's bytes.
+ * @param[in] source The span containing the bytes to copy to the destination.
+ * @return An #az_span that is a slice of the \p destination span (i.e. the remainder) after the
+ * source bytes has been copied.
+ *
+ * @remarks The method assumes that the \p destination has a large enough size to hold the \p
+ * source.
+ * @remarks This method copies all of source to destination even if source and destination overlap.
  */
-az_span az_span_append_uint8(az_span destination, uint8_t byte);
+az_span az_span_copy(az_span destination, az_span source);
 
 /**
- * @brief az_span_append appends the bytes referred to by the source span into the
- * destination span starting at the destination span length.
+ * @brief az_span_copy_uint8 copies the uint8 \p byte to the \p destination at its 0-th index.
  *
- * @param[in] destination The az_span where the bytes should be appended to.
- * @param[in] source Refers to the bytes to be appended to the destination
- * @return An #az_span that is a clone of the \p destination span with its length increased by the
- * \p source length. The method assumes that the \p destination has a large enough capacity to hold
- * the \p source.
+ * @param[in] destination The az_span where the byte should be copied to.
+ * @param[in] byte The uint8 to copy into the destination span.
+ * @return An #az_span that is a slice of the \p destination span (i.e. the remainder) after the
+ * byte has been copied. The method assumes that the \p destination has a large enough size to
+ * hold one more byte.
  */
-az_span az_span_append(az_span destination, az_span source);
+az_span az_span_copy_uint8(az_span destination, uint8_t byte);
```

Comment:
Rename this function az_span_copy_u8 (to match i32, u64, etc.)

Reply:
Sounds good. How about the word `copy`. Should this instead be `az_span_write_u8`?

If we approach it like follows, `copy` makes sense:
az_span_copy - copies a span of bytes into the destination
az_span_copy_u8 - copies a single byte into the destination

It also has the benefit of conveying what the signature means (returning remainder/etc.) and we reduce the number of verbs that callers need to think about (consistent call sites, effectively method overloading).

`write` seems to be more accurate in what we are doing in the method and what the intention of the caller is. We are writing a single byte to a particular memory address.